src/main/java/org/commcare/suite/model/Global.java

```diff
@@ -0,0 +1,48 @@
+package org.commcare.suite.model;
+
+import org.javarosa.core.util.ArrayUtilities;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.ExtUtil;
+import org.javarosa.core.util.externalizable.ExtWrapList;
+import org.javarosa.core.util.externalizable.Externalizable;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Vector;
+
+/**
+ * Defines top level UI logic for a case-select or case-detail view,
+ * Part of {@code Detail} model
+ */
+public class Global implements Externalizable {
+
+    private GeoOverlay[] geoOverlays;
+
+    /**
+     * Serialization Only
+     */
+    public Global() {
+    }
+
+    public Global(Vector<GeoOverlay> geoOverlays) {
+        this.geoOverlays = ArrayUtilities.copyIntoArray(geoOverlays, new GeoOverlay[geoOverlays.size()]);
+    }
+
+    @Override
+    public void readExternal(DataInputStream in, PrototypeFactory pf) throws IOException, DeserializationException {
+        Vector<GeoOverlay> theGeoOverlays = (Vector<GeoOverlay>)ExtUtil.read(in, new ExtWrapList(GeoOverlay.class), pf);
+        geoOverlays = new GeoOverlay[theGeoOverlays.size()];
+        ArrayUtilities.copyIntoArray(theGeoOverlays, geoOverlays);
+    }
+
+    @Override
+    public void writeExternal(DataOutputStream out) throws IOException {
+        ExtUtil.write(out, new ExtWrapList(ArrayUtilities.toVector(geoOverlays)));
```

Comment:
Does this work if the list is empty/null?

I think there's a good chance we'll be using a global static namespace for detail info quite soon, and we'll want the model to be pretty flexible.

Reply:
`GlobalParser` today always [initialize this list](https://github.com/dimagi/commcare-core/blob/geoOverlay/src/main/java/org/commcare/xml/GlobalParser.java#L26) so that it's never null. Would we still want to provide null safety here ?

Reply:
Nope, no need as long as it works (sometimes it's hard to tell with null boundary checks.

It did make me realize, though, that there aren't any sample app / parse tests for all of this. Can you write some up? Nothing complicated, just making sure the code paths get hit, and demonstrating the boundaries.